src/auth0/clients.js

```diff
@@ -0,0 +1,267 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const ValidationError = require('auth0-extension-tools').ValidationError;
+
+const utils = require('../utils');
+// const constants = require('../constants');
+
+/*
+ * Get all non-global clients
+ */
+const getClients = function(progress, client) {
+  if (progress.clients) {
+    return Promise.resolve(progress.clients);
+  }
+
+  /* Grab all non-global clients */
+  return Promise.all(client.clients.getAll({ global: false }))
+    .then(function(allClients) {
+      progress.clients = _.chain(allClients)
+        .flattenDeep()
+        .union()
+        .value();
+      return progress.clients;
+    });
+};
+
+const updateExistingClient = function(progress, client, clientName, clientConfig, existingClient) {
+  /* Get entire client config */
+  clientConfig = utils.parseJsonFile(clientName, clientConfig.scriptFile);
```

Comment:
does `scriptFile` make sense here?  can we rename to something else? like `settingsFile` `configurationFile` ... ?